### PR TITLE
Resolve user defined functions against the search path.

### PR DESCRIPTION
**Pull request by andreidan**
## Checklist

 - [x] User relevant changes are recorded in ``CHANGES.txt``
 - [x] Touched code is covered by tests
 - [x] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

**Comment by andreidan** [comment]
retest this please

File: sql/src/main/java/io/crate/metadata/Functions.java [hunk 1]
```diff
@@ -148,35 +149,78 @@ public FunctionImplementation getUserDefined(String schema,
...
+     * @param searchPath The {@link SearchPath} against which to try to resolve the function if it is not identified by
+     *                   a fully qualifed name (ie. `schema.functionName`)
      * @return a function implementation.
      */
     @Nullable
-    public FunctionImplementation getUserDefinedByArgs(String schema,
-                                                       String name,
-                                                       List<? extends FuncArg> arguments) throws UnsupportedOperationException {
-        FunctionResolver resolver = lookupUdfFunctionResolver(schema, name, arguments);
+    public FunctionImplementation resolveUserDefinedByArgs(QualifiedName functionQualifiedName,
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/metadata/Functions.java]: couldn't we keep schema and name as arguments here? it looks like we're creating QualifiedName right at the call site just to unpack it immediately

Comment (andreidan) [reply sql/src/main/java/io/crate/metadata/Functions.java]: Yes we can, was looking to keep the `resolve...` methods consistent

**Comment by andreidan** [comment]
retest this please

**Pull request merged**